pkg/daemon/drain.go

```diff
@@ -153,26 +152,40 @@ func (dn *Daemon) performDrain() error {
 	return nil
 }
 
-// isDrainRequired determines whether node drain is required or not to apply config changes.
-func isDrainRequired(actions, diffFileSet []string, oldIgnConfig, newIgnConfig ign3types.Config) (bool, error) {
+type ReadFileFunc func(string) ([]byte, error)
+
+// drainIfRequired determines whether node drain is required or not to apply config changes and performs the drain if required
+// returns whether drain was attempted
+func (dn *Daemon) drainIfRequired(actions, diffFileSet []string, readOldFile, readNewFile ReadFileFunc) (bool, error) {
+	// For any unhandled cases (although currently there aren't any), default to drain
+	drain := true
 	if ctrlcommon.InSlice(postConfigChangeActionReboot, actions) {
```

Comment:
I don't think it's actually redundant because technically actions is a list (although currently it's only ever of length 1). So if we didn't have this if branch, if actions included both postConfigChangeActionReboot and postConfigChangeActionReloadCrio we could take eg the postConfigChangeActionReloadCrio branch and skip a drain